src/main/java/io/horrorshow/codey/time/DiscordTimezone.java

```diff
@@ -0,0 +1,135 @@
+package io.horrorshow.codey.time;
+
+import io.horrorshow.codey.discordutil.DiscordUtils;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.Arrays;
+
+@Service
+public class DiscordTimezone extends ListenerAdapter {
+    private final DiscordUtils utils;
+
+    public DiscordTimezone(@Autowired JDA jda, @Autowired DiscordUtils utils) {
+        this.utils = utils;
+
+        jda.addEventListener(this);
+    }
+
+    private final String emoji = "⏰";
+    private final String deleteEmoji = "\uD83D\uDDD1";
+
+    @Override
+    public void onGuildMessageReceived(@NotNull GuildMessageReceivedEvent event) {
+        if (!event.getAuthor().isBot()) {
+            onMessage(event.getMessage());
+        }
+    }
+
+    private void onMessage(Message message) {
+        String rawContent = message.getContentRaw();
+        String[] rawContentParts = rawContent.split("\\s+");
+
+        if(rawContentParts.length == 0)
+            return;
+
+        Object[] result = time(rawContentParts);
+
+        if(!((boolean) result[0])) {
+            if(result[1] != null && (boolean) result[1])
+            {
+                message.addReaction(emoji).complete();
+                message.reply(new EmbedBuilder()
+                        .setColor(utils.getColor())
+                        .setTitle("Did you know that I can provide time conversions automatically?")
+                        .addField("Here is a tip:",
+                                "You can use regions from the IANA Time Zone Database (TZDB). " +
+                                "This has region IDs of the form `{area}/{city}`, such as `Europe/Paris` or `America/New_York`", false)
+                        .build()).complete().addReaction(deleteEmoji).complete();
+            }
+            return;
+        }
+
+        OffsetDateTime timeStamp = (OffsetDateTime) result[3];
+
+        message.reply(new EmbedBuilder()
+                .setFooter("Time in your timezone: ")
+                .setTimestamp(timeStamp)
+                .setColor(utils.getColor())
+                .build()).mentionRepliedUser(false).complete();
+    }
+
+    private Object[] time(String[] rawContentParts) {
```

Comment:
I suggest splitting all this logic into several methods where each one has only one responsibility:
 1. a method that finds all times in a string (the discord message) and returns it (This method needs some unit tests with sample strings to make it clear how it behaves and verify that it's not broken when adding features / changes to it later)
 2. parsing the time string from the message into the java.time format
 3. handling interaction with discord (like sending messages, etc.) with the result from 2.